### PR TITLE
Fpath fd fixes

### PR DESCRIPTION
**Pull request by whippit**
updated to be more efficient and fix file descriptor leaks.

### CODE REVIEW

File: synapse/lib/filepath.py [hunk 1]
```diff
@@ -327,7 +470,7 @@ def _pathClass(*paths):
         raise s_exc.NoSuchPath(path=path)
 
     if os.path.isdir(path):
-        return path_ctors.get('fs.reg.dir')
+        return path_ctors.get('fs.reg.file')
```
Comment (vertexmc) [inline synapse/lib/filepath.py]: Should this still be `fs.reg.dir`?

File: synapse/lib/filepath.py [hunk 2]
```diff
@@ -500,9 +656,8 @@ def isfile(*paths):
 
     if not fpath:
         return False
-    if fpath.isfile():
-        return True
-    return False
+    fpath.close()
```
Comment (vertexmc) [inline synapse/lib/filepath.py]: Why close here?

File: synapse/lib/filepath.py [hunk 3]
```diff
@@ -514,26 +669,23 @@ def isdir(*paths):
 
     if not fpath:
         return False
-    if fpath.isfile():
-        return False
-    return True
+    fpath.close()
```
Comment (vertexmc) [inline synapse/lib/filepath.py]: Why close here?

File: synapse/lib/filepath.py [hunk 4]
```diff
@@ -514,26 +669,23 @@ def isdir(*paths):
...
     '''
     Determines if the path exists even if the path terminates inside a container file.
     If a list of paths are provided, they are joined first.
     Returns a boolean.
     '''
-    if not parsePath(*paths):
+    fpath = parsePath(*paths)
+    if not fpath:
         return False
+    fpath.close()
```
Comment (vertexmc) [inline synapse/lib/filepath.py]: Why close here?

File: synapse/lib/filepath.py (hunk 1 above)
Comment (whippit) [inline synapse/lib/filepath.py]: regular filesystem files and dirs are addressed by a singular class now.

File: synapse/lib/filepath.py (hunk 2 above)
Comment (whippit) [inline synapse/lib/filepath.py]: There may be spooled temp files still open.  Try running the tests with that line commented out ;)

**Pull request merged**